calyx-ir/src/guard.rs

```diff
@@ -21,24 +32,53 @@ pub enum PortComp {
 }
 
 /// An assignment guard which has pointers to the various ports from which it reads.
-#[derive(Default, Debug, Clone)]
-pub enum Guard {
-    #[default]
-    /// The constant true
-    True,
+#[derive(Debug, Clone)]
+pub enum Guard<T> {
     /// Represents `c1 || c2`.
-    Or(Box<Guard>, Box<Guard>),
+    Or(Box<Guard<T>>, Box<Guard<T>>),
     /// Represents `c1 && c2`.
-    And(Box<Guard>, Box<Guard>),
+    And(Box<Guard<T>>, Box<Guard<T>>),
     /// Represents `!c1`
-    Not(Box<Guard>),
+    Not(Box<Guard<T>>),
+    /// The constant true
+    True,
     /// Comparison operator.
     CompOp(PortComp, RRC<Port>, RRC<Port>),
     /// Uses the value on a port as the condition. Same as `p1 == true`
     Port(RRC<Port>),
+    /// Other types of information.
+    Info(T),
+}
+
+impl<T> Default for Guard<T> {
+    fn default() -> Self {
+        Guard::True
+    }
 }
 
-impl Hash for Guard {
+#[derive(Debug, Clone)]
+pub struct StaticTiming {
+    interval: (u64, u64),
+    parent: RRC<Group>,
+}
+
+impl ToString for StaticTiming {
+    fn to_string(&self) -> String {
+        let mut full_string = "%".to_owned();
+        full_string.push_str(self.parent.borrow().name().as_ref());
+        full_string.push('[');
+        full_string.push_str(&self.interval.0.to_string());
+        full_string.push(':');
+        full_string.push_str(&self.interval.1.to_string());
+        full_string.push(']');
+        full_string
```

Comment:
Use the `format!` macro instead of manually pushing things into the string buffer